distributed/executor.py

```diff
@@ -975,6 +976,36 @@ def cancel(self, futures):
         return sync(self.loop, self._cancel, futures)
 
     @gen.coroutine
+    def _publish_dataset(self, keys, sdata, name):
+        yield self.scheduler.publish_data(keys=keys, data=sdata, name=name)
+
+    def publish_dataset(self, data, name):
+        """
+        Store a named reference to the data in the scheduler for other executors
+
+        Parameters
+        ----------
+        data: list of futures or dask collection
+        name: str
+            handle by which the stored data should be known
+
+        Returns
+        -------
+        None
+        """
+        with serialize_for_publish(data) as out:
+            sdata, keys = out
+            self._send_to_scheduler({'op': 'publish-data', 'keys': keys,
+                                     'data': sdata, 'name': name})
+
+    @property
+    def published_datasets(self):
+        if hasattr(self, '_loop_thread'):
```

Comment:
I saw it elsewhere - should you not check that the scheduler is around and listening before trying to pull a response? In any case, this doesn't work yet.